Assets/Scripts/Model/Weapon.cs

```diff
@@ -32,13 +40,46 @@ void Update ()
 
 	}
 
+	// Upgrade weapon features
 	public void Upgrade ()
 	{
 		this.currentDamage *= upgradeFactor;
 	}
+		
+	//USeed for sounds
+	void Awake()
+	{
+		this.source = GetComponent<AudioSource>();
+	}
 
-    /*public float getDamage()
+	// Getters
+    public float getCurrentDamage()
     {
         return this.currentDamage;
-    }*/
+    }
+
+	public void addTarget(CanReceiveDamage target){
+		this.targets.Add (target);
+		Debug.Log ("Target in queue");
+	}
+
+	// Called to attack a target
+	public void Attack()
```

Comment:
It's not an attribute of the weapon, the list is ... I'm not sure to understand what you mean. Explain please.